src/components/Output.tsx

```diff
@@ -268,16 +271,68 @@ export const Output = ({
                 valueBN.toFormat(fractionDigits ?? 0, roundingMode, {
                   ...format,
                 }),
-              [OutputType.Fiat]: () =>
-                valueBN.toFormat(fractionDigits ?? USD_DECIMALS, roundingMode, {
-                  ...format,
-                  prefix: '$',
-                }),
-              [OutputType.SmallFiat]: () =>
-                valueBN.toFormat(fractionDigits ?? SMALL_USD_DECIMALS, roundingMode, {
-                  ...format,
-                  prefix: '$',
-                }),
+              [OutputType.Fiat]: () => {
+                const formattedValue = valueBN.toFormat(
+                  fractionDigits ?? USD_DECIMALS,
+                  roundingMode,
+                  {
+                    ...format,
+                    prefix: '$',
+                  }
+                );
+
+                const { significantDigits, decimalDigits, zeros } = formatZeroNumbers(
+                  formattedValue,
+                  true
+                );
+
+                if (compressZeros) {
+                  return (
+                    <div>
+                      {significantDigits}.
+                      {Boolean(zeros) && (
+                        <>
+                          0<Styled.Sub title={formattedValue}>{zeros}</Styled.Sub>
+                        </>
+                      )}
+                      {decimalDigits}
+                    </div>
+                  );
+                }
```

Comment:
nit: abstract to a function so it can be re-used for all `OutputTypes`

Reply:
Ye ofc, do you mean if I can create a reusable tsx component? We can make it available for more output types for compatible ones (That is, all the numeric ones I would say)

Reply:
It can be a fn within Output or a separate component. Trust your judgement here.

Reply:
Done